cdap-ui/app/cdap/components/PluginSchemaEditor/index.tsx

```diff
@@ -373,6 +396,23 @@ class PluginSchemaEditorBase extends React.PureComponent<
   };
 
   private santizeSchemasForEditor = (schemas = this.props.schemas): ISchemaType[] => {
+    if (typeof schemas === 'string') {
+      let returnSchema;
+      try {
+        returnSchema = JSON.parse(schemas);
+      } catch (e) {
+        return [{ ...getDefaultEmptyAvroSchema() }];
+      }
+      if (!returnSchema.schema) {
+        return [
+          {
+            name: 'etlSchemaBody',
+            schema: !returnSchema.fields.length ? getDefaultEmptyAvroSchema() : returnSchema,
```

Comment:
Nit: In other places, I've seen a preference for putting the "truthy" condition earlier in the ternary (i.e., this would be `!!returnSchema.fields.length ? returnSchema : getDefaultEmptyAvroSchema()`). Do we have a preference here?

Reply:
IMHO there is no preference as of now. But if you insist on a pattern we could follow. I am not too strong about it.

Reply:
I think it makes things a bit easier to read, but is most effective with nested ternaries (which we shouldn't be doing anyway). So I'm not going to push for a standard for now.

Reply:
I think I was using that pattern in Replication if I were to use the `<If condition>`, since condition has to be boolean